[RHACS] Added release notes for 4.8 (#95013)
openshift/openshift-docs: release_notes/48-release-notes.adoc

+
For more details, see xref:../operating/examine-images-for-vulnerabilities.adoc#accessing-delegated-image-scanning_examine-images-for-vulnerabilities[Accessing delegated image scanning].

* In `roxctl` CLI, certificate validation failures are now marked as errors.

* {product-title-short} 4.8 includes the updated `roxctl` help command output making it more readable. The output is now more consistent with other command-line tools.

* Red Hat has moved the `SecurityPolicy`` Custom Resource Definition (CRD) to the template directory within the Helm chart. This change simplifies CRD maintenance if you are using Helm, as it will now automatically upgrade.
+
[IMPORTANT]
====
If you are using Helm to manage your {product-title-short} installation, you must apply the following changes to the `SecurityPolicy` CRD before upgrading to avoid upgrade failures:
[source,terminal]
----
$ kubectl annotate crd/securitypolicies.config.stackrox.io meta.helm.sh/release-name=stackrox-central-services <1>
$ kubectl annotate crd/securitypolicies.config.stackrox.io meta.helm.sh/release-namespace=stackrox <2>
$ kubectl label crd/securitypolicies.config.stackrox.io app.kubernetes.io/managed-by=Helm
----

> <1> Update the `release-name` annotation to match the name if you used different name during your initial installation. The default value is `stackrox-central-services`.

Reviewer: For this footnote, and the next, put the "if" statement first. This makes it easier for users to decide if they need to read the rest of the sentence.

--

[id="bug-fixes_{context}"]
== Bug fixes in version 4.8.0

*Release date*: 23 June 2025

> * Previously, the Secured Cluster sensor would remain uninitialized and offline if messages contained non-UTF-8 characters.

Reviewer: Put "if" statement first.

* Previously, creating a security policy with a cluster scope using the cluster's name would cause the UI to crash upon viewing the policy.
It was due to the system's inability to resolve the cluster name to its corresponding ID correctly.
This update enables proper resolution of cluster names to IDs in security policies.
As a result, you can now view policies with cluster scope in the UI without encountering errors.

* Previously, the Scanner V4 failed to identify some critical CVEs in Java workloads because an `unidentified jar` error caused the scanner to skip valid JAR files during the scanning process.
As a consequence, RHACS did not detect these vulnerabilities in the scan results.
This update eliminates the `unidentified jar" error, enabling the scanner to process JAR files properly.
As a result, the Scanner V4 now accurately identifies critical CVEs in Java workloads, providing comprehensive vulnerability scanning.

* Previously, the **Cancel** button on the delegated scanning page provided no visual feedback if you made no changes, leading to confusion about its functionality.

> It was because the button only reset the form for unpersisted changes.

Reviewer: ```suggestion
This lack of feedback occurred because the button only reset the form for unpersisted changes.
```
